### PR TITLE
Add page explaining the image diff api

### PR DESCRIPTION
**Pull request by jamonation**
## Type of change
Documentation

### What should this PR do?
Add a page explaining how to use the Image Diff API

### Why are we making this change?
We need docs on the Image Diff API

### What are the acceptance criteria? 
Tutorial has already been reviewed 

### How should this PR be tested?
N/A

### CODE REVIEW

File: content/chainguard/chainguard-images/using-the-image-diff-api.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  . . .
+}
+}
+```
+
+In this example output, the list of `added` packages include things like `bash` and other helpful utilities that are package in the nginx development image. If there are any vulnerabilities detected in an image they will be listed in that section of the output.
+
+## Making a Request Programmatically to the Diff API Using Go
+
+If you would like to query the Diff API programmatically, there is an example Go application hosted in our [platform-examples](https://github.com/chainguard-dev/platform-examples/tree/main/image-diff) GitHub repository. The example uses the [Chainguard SDK](https://pkg.go.dev/chainguard.dev/sdk) and is a good way to learn how to interact with the API programmatically.
```
Comment (SharpRake) [inline content/chainguard/chainguard-images/using-the-image-diff-api.md:166]:
```suggestion
If you would like to query the Diff API programmatically, there is an example Go application hosted in our [platform-examples](https://github.com/chainguard-dev/platform-examples/tree/main/image-diff) GitHub repository. The example uses the [Chainguard SDK](https://pkg.go.dev/chainguard.dev/sdk) and is a good way to learn how to interact with the API through a declarative approach.
```

I felt like "programmatically was feeling kind of repetitive

File: content/chainguard/chainguard-images/using-the-image-diff-api.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+Clone the repository, or copy the code manually and set some shell variables. Again we're using the `cgr.dev/chainguard/nginx` image as an example:
```
Comment (SharpRake) [inline content/chainguard/chainguard-images/using-the-image-diff-api.md:168]:
```suggestion
Clone the repository, or copy the code manually. Again we're using the `cgr.dev/chainguard/nginx` image as an example:
```

File: content/chainguard/chainguard-images/using-the-image-diff-api.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+```sh
+git clone https://github.com/chainguard-dev/platform-examples
+cd platform-examples/image-diff
+```
+
```
Comment (SharpRake) [inline content/chainguard/chainguard-images/using-the-image-diff-api.md:174]:
```suggestion

Following that, set some shell variables:

```

Didn't make sense to me to have two separate code blocks like this

File: content/chainguard/chainguard-images/using-the-image-diff-api.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+menu:
+  docs:
+    parent: "chainguard-images"
+weight: 450
+toc: true
+---
+
+## Introduction to the Chainguard Images Diff API
+
+The Chainguard Images Diff API generates a JSON formatted list of package additions, removals, and detected CVEs in Chainguard Images. It is intended to provide an authenticated, machine readable, formatted list of changes between two image builds. Combined with the [Tag History API](/chainguard/chainguard-images/using-the-tag-history-api/), you can use the Diff API to compare arbitrary builds of an image over time and programmatically detect changes that might require gating a release or modifying your application code.
```
Comment (SharpRake) [inline content/chainguard/chainguard-images/using-the-image-diff-api.md:19]:
```suggestion
The Chainguard Images Diff API generates a JSON-formatted list of package additions, removals, and detected CVEs in Chainguard Images. It is intended to provide an authenticated, machine readable, formatted list of changes between two image builds. Combined with the [Tag History API](/chainguard/chainguard-images/using-the-tag-history-api/), you can use the Diff API to compare arbitrary builds of an Image over time and programmatically detect changes that might require gating a release or modifying your application code.
```

File: content/chainguard/chainguard-images/using-the-image-diff-api.md [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
```
Comment (SharpRake) [inline content/chainguard/chainguard-images/using-the-image-diff-api.md:20]: The diff API only works with Chainguard Images, right? I might make mention of that somewhere

File: content/chainguard/chainguard-images/using-the-image-diff-api.md [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+This guide will cover:
+
+1. How to get an authentication token to query the Diff API
+2. Identifying image digests in a repository for comparison
```
Comment (SharpRake) [inline content/chainguard/chainguard-images/using-the-image-diff-api.md:24]: If the diff api only works with Chainguard Images, you might want to go through and change any `image` to `Image` to align with our product messaging

File: content/chainguard/chainguard-images/using-the-image-diff-api.md [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+{"code":16, "message":"failed to verify token", "details":[]}
+```
+
+For the purposes of this guide, we will use the `public-images` Chainguard IAM group. Set an environment variable corresponding to the group's [UIDP](/chainguard/administration/cloudevents/events-reference/#uidp-identifiers), which will be used later on:
+
+```sh
+GROUP_UIDP=$(chainctl iam group describe public-images -ojson |jq -r '.id')
+```
+
+If you are using a private registry, be sure to substitute in your IAM group name in place of `public-images` to use your specific private group UIDP identifier in your subsequent commands.
```
Comment (SharpRake) [inline content/chainguard/chainguard-images/using-the-image-diff-api.md:67]:
```suggestion
If you are using a private registry, be sure to substitute your IAM group name in place of `public-images` to use your specific private group UIDP identifier in your subsequent commands.
```

File: content/chainguard/chainguard-images/using-the-image-diff-api.md [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+
+The first digest in this example will be referred to as the `FROM_DIGEST`, and the second comparison image as `TO_DIGEST`.
+
+Begin by creating a shell variable to hold the repository URL:
+
+```sh
+REPO_URL="cgr.dev/chainguard/nginx"
+```
+
+Note: if you are using a private registry, replace `cgr.dev/chainguard/nginx` with your relevant private Chainguard Registry and repository name as required.
```
Comment (SharpRake) [inline content/chainguard/chainguard-images/using-the-image-diff-api.md:81]:
```suggestion
If you are using a private registry, be sure to replace `cgr.dev/chainguard/nginx` with your relevant private Chainguard Registry and repository name as required.
```

I think it looks nicer if we only use "Note:" in places where we do a proper note block (with a `>` preceding it)

File: content/chainguard/chainguard-images/using-the-image-diff-api.md [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+
+Now you can run the following command to record the repository UIDP identifier as a variable:
+
+```sh
+REPO_UIDP=$(curl -s -X GET -H "Authorization: Bearer $TOKEN" \
+  "https://console-api.enforce.dev/registry/v1/repos?name=${REPO_NAME}" | \
+  jq -r ".items[] | select(.id | startswith(\"${GROUP_UIDP}\")) | .id")
+```
+
+Now with all of these variables set, you can call the Diff API with the following command:
```
Comment (SharpRake) [inline content/chainguard/chainguard-images/using-the-image-diff-api.md:124]:
```suggestion
With all of these variables set, you can call the Diff API with the following command:
```

**Review by SharpRake — APPROVED** [review]
Tech test worked like a charm on this one, great work! All my comments are rather nitpicky - let me know if you have any questions about them.

File: content/chainguard/chainguard-images/using-the-image-diff-api.md (hunk 5 above)
Comment (jamonation) [reply content/chainguard/chainguard-images/using-the-image-diff-api.md:19]: Right, added another Chainguard Images, thanks for the suggestion!

File: content/chainguard/chainguard-images/using-the-image-diff-api.md (hunk 6 above)
Comment (jamonation) [reply content/chainguard/chainguard-images/using-the-image-diff-api.md:23]: Done, good call out

**Pull request merged**